Indicators/Variance.cs

```diff
@@ -78,7 +78,8 @@ protected override decimal ComputeNextValue(IReadOnlyWindow<IndicatorDataPoint>
                 _rollingSumOfSquares -= removedValue * removedValue;
             }
 
-            return meanValue2 - meanValue1 * meanValue1;
+            // Ensure non-negative variance
+            return System.Math.Max(0m, meanValue2 - meanValue1 * meanValue1);
```

Comment:
From my understanding, Variance should never be negative since it squares the difference. So, I'm wondering if the result being negative is a result of this progressive algorithm carrying a precision error? or how is it we ended up with a negative number for the Variance? I think it would be nice to confirm/note this

Reply:
Right, it is a matter of precision error. It led `meanValue1 * meanValue1` being greater than `meanValue2` by a very small negative number: `-2e-25`.